realtek: RTL930x/RTL931x led_set defines, and led-set count debug print tidy-up

bevanweiss opened the pull request:
Add defines for RTL930x and RTL931x led_set 'modes' (to avoid magic numbers in dts files).

Also tidy up the handling of led_sets on rtl930x devices (so it doesn't produce errors for missing led_sets)

Tested-on: Hasivo S1100WP-8GT-SE, RTL9303, 3xHC595 chips for LEDs
Signed-off-by: Bevan Weiss <bevan.weiss@gmail.com>

bevanweiss committed:
realtek: RTL930x/RTL931x led_set defines

Add defines for RTL930x and RTL931x led_set 'modes' (to avoid magic numbers
in dts files).

Signed-off-by: Bevan Weiss <bevan.weiss@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19791
Signed-off-by: Robert Marko <robimarko@gmail.com>

plappermaul commented:
No LED specialist here. So first of all is this aligned with https://github.com/openwrt/openwrt/pull/19573 ?

bevanweiss commented on target/linux/realtek/files-6.12/drivers/net/dsa/rtl83xx/rtl931x.c:
```diff
@@ -25,6 +25,8 @@
 #define RTL931X_VLAN_PORT_TAG_ITPID_IDX_MASK			GENMASK(2,1)
 #define RTL931X_VLAN_PORT_TAG_ITPID_KEEP_MASK			GENMASK(0,0)
 
+#define RTL930X_GPIO_SEL_GPIO31_LEDSYNC_SEL 			BIT(16)
```
Thanks, fixed... too much copy-pasta from me

bevanweiss commented:
The `led-sync` needs a change to the dts yaml file... but I'm not seeing that in the openwrt repo...
What's the 'right way' to tie this doco change in?

jonasjelonek commented:
If such a file is added by a patch, you'd add another patch on top for this. But I don't see such a patch, so I guess there are no dt-bindings for this, and there's nothing you could change (aka cannot fix non-existent docs) ^^

bevanweiss commented:
> If such a file is added by a patch, you'd add another patch on top for this. But I don't see such a patch, so I guess there are no dt-bindings for this, and there's nothing you could change (aka cannot fix non-existent docs) ^^

Unfortunate, but good point.
I've just searched upstream linux for the `realtek,rtl838x-eth` compatible, with no hits... so I guess it is only an OpenWRT downstream thing in its entirety right now.

Would it be good to have something?  If so, I'll look to put something together next weekend.

jonasjelonek commented:
The driver has been (or still is?) in a never-upstreamable state for a long time, so this doesn't exist upstream. And there was not really a point of having the dt-bindings defined for it then.

IMO this should be done in combination with an attempt to get the driver(s) upstream when we have it in a suitable state. But not now. According to @plappermaul plans some DT related things might change a lot (correct me if I'm wrong), and we do not really make use of dt-bindings downstream.

plappermaul commented on target/linux/realtek/files-6.12/drivers/net/dsa/rtl83xx/rtl930x.c:
```diff
@@ -2272,11 +2274,20 @@ static void rtl930x_led_init(struct rtl838x_switch_priv *priv)
 		}
 
+		if (leds_in_this_set <= 0 || leds_in_this_set > sizeof(set_config)) {
+			pr_err("%s: %s configuration invalid skipping over this set, leds_in_this_set=%d (should be >0 and <=%d\n",
+				__func__, set_name, leds_in_this_set, sizeof(set_config));
```
Please use dev_err/dev_info instead of pr_XXX("%s ...", __func__);

bevanweiss replied:
I'll update all the references in this function.

plappermaul commented on target/linux/realtek/files-6.12/drivers/net/dsa/rtl83xx/rtl930x.c:
```diff
@@ -2312,6 +2323,12 @@ static void rtl930x_led_init(struct rtl838x_switch_priv *priv)
 	sw_w32_mask(0x3, 0x1, RTL930X_LED_GLB_CTRL);
 
+	/* Enable GPIO19_LEDSYNC if utilised */
+	if (of_property_read_bool(node, "led-sync"))
```
We are currently using "led_set" (underscore) no need for this deviation. Only if you fix that later.

bevanweiss replied:
> We are currently using "led_set" (underscore) no need for this deviation. Only if you fix that later.

My understanding was that the linux kernel prefers kebab-case property names in the device tree, and this `led-sync` aligns with the `active-low` below
```c
if (of_property_read_bool(node, "active-low"))
```

achterin commented on target/linux/realtek/files-6.12/drivers/net/dsa/rtl83xx/rtl930x.c:
```diff
@@ -2324,7 +2342,7 @@ static void rtl930x_led_init(struct rtl838x_switch_priv *priv)
 
 	for (int i = 0; i < 24; i++)
-		pr_debug("%s %08x: %08x\n",__func__, 0xbb00cc00 + i * 4, sw_r32(0xcc00 + i * 4));
+		dev_dbg(dev, "%s %08x: %08x\n",__func__, 0xbb00cc00 + i * 4, sw_r32(0xcc00 + i * 4));
```
```suggestion
		dev_dbg(dev, "%s %08x: %08x\n", __func__, 0xbb00cc00 + i * 4, sw_r32(0xcc00 + i * 4));
```

ecsv reviewed (changes requested):
Macros with complex values should be enclosed in parentheses

plappermaul reviewed (changes requested):
Thanks.

bevanweiss committed:
realtek: RTL930x led_set count error message tidy up + dev_ print fixups

Whilst testing Hasivo s1100wp-8gt-se LED configuration, several error
messages were presented which didn't indicate which led_set they were
referencing, nor what the value was that caused the invalid configuration.

Migrate to use dev_ print messages for this function.
And tidy up both when the error message is reported (don't show it when
an led_set isn't in the DTS) and what details the message presents.

Signed-off-by: Bevan Weiss <bevan.weiss@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19791
Signed-off-by: Robert Marko <robimarko@gmail.com>

plappermaul commented on target/linux/realtek/files-6.12/drivers/net/dsa/rtl83xx/rtl930x.c:
```diff
@@ -2247,12 +2247,13 @@ static void rtl930x_set_distribution_algorithm(int group, int algoidx, u32 algom
 	u32 pm = 0;
 
-	pr_debug("%s called\n", __func__);
+	dev_dbg(dev, "%s called\n", __func__);
```
As said. drop this.

plappermaul commented on target/linux/realtek/files-6.12/drivers/net/dsa/rtl83xx/rtl930x.c:
```diff
@@ -2247,12 +2247,13 @@ static void rtl930x_set_distribution_algorithm(int group, int algoidx, u32 algom
 	node = of_find_compatible_node(NULL, NULL, "realtek,rtl9300-leds");
 	if (!node) {
-		pr_debug("%s No compatible LED node found\n", __func__);
+		dev_dbg(dev, "%s No compatible LED node found\n", __func__);
```
and remove __func__ from others.

plappermaul reviewed (changes requested):
Ok with this series afterwards.
